src/components/Login.jsx

```diff
@@ -1,193 +1,186 @@
-import React, {Component} from "react";
-import {browserHistory} from "react-router";
-import {Avatar, Box, Button, Divider, GridList, GridListTile, Paper, TextField, Typography, Link} from "@material-ui/core";
+import React from "react";
+import { useDispatch, useSelector } from "react-redux";
+import { browserHistory } from "react-router";
+import {
+	Avatar,
+	Box,
+	Button,
+	Divider,
+	ImageList,
+	ImageListItem,
+	Paper,
+	TextField,
+	Typography,
+	Link
+} from "@material-ui/core";
 import LockOutlinedIcon from "@material-ui/icons/LockOutlined";
 import Version from "./children/Version";
 import Cookies from "universal-cookie";
-import {withStyles} from "@material-ui/core/styles";
+import { makeStyles } from "@material-ui/core/styles";
 import config from "../app.config";
 
+import { login } from "../actions";
+
 const cookies = new Cookies();
-const styles = theme => ({
-	resetPW:{
-		display:"block",
-		textAlign:"right",
+
+const useStyles = makeStyles((theme) => ({
+	resetPW: {
+		display: "block",
+		textAlign: "right",
 		margin: "0 auto 10px auto",
 		fontFamily: theme.typography.body1.fontFamily
 	},
-	tos:{
+	tos: {
 		fontSize: "12px",
 		fontFamily: theme.typography.body2.fontFamily,
-		margin:"20px auto 10px auto"
+		margin: "20px auto 10px auto"
 	},
-	signUp:{
+	signUp: {
 		fontWeight: 500,
-		fontSize:"15px",
+		fontSize: "15px",
 		fontFamily: theme.typography.body1.fontFamily,
-		display:"block",
-		margin:"10px auto 5px auto"
+		display: "block",
+		margin: "10px auto 5px auto"
 	}
-});
-
-
-
-class Login extends Component {
-	constructor(props) {
-		super(props);
-
-		this.state = {
-			username: "",
-			password: "",
-			passwordErrorText: "",
-			loginErrorText: "",
-			error: false,
-			origin: props.location.query["origin"],
```

Comment:
you might want to keep this origin;
it's used to redirect to the previous viewer you were on before login

Reply:
Oh, I see! It was giving a lot of errors on login errors so I thought since it is redirecting to the main page ill do it myself. I didn't think it was for redirecting to the previous component. I'll change this. Thanks!